### PR TITLE
collections panel refactor

### PR DESCRIPTION
**Pull request by tsop14**
Creates parent directive grCollectionsTree which takes the attributes needed to construct grNodes and grNode. 

grNodes and grNode reference this parent directive for unchanging attributes - rather than setting them again in scope object. 

Adds ability to add a custom select event - for future non-navigation uses of the collections tree.  🌲

### CODE REVIEW

File: kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js [hunk 1]
```diff
@@ -99,75 +98,82 @@ grCollectionsPanel.controller('GrNodeCtrl',
...
     return {
         restrict: 'E',
+        require: ['grNode', '^^grCollectionTree'],
         scope: {
             node: '=grNode',
-            nodeList: '=grNodeList',
-            editing: '=grEditing',
-            selectedImages$: '=grSelectedImages',
-            selectedCollections: '=grSelectedCollections'
+            nodeList: '=grNodeList'
```
Comment (theefer) [inline kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js]: Is this not just `grCollectionTreeCtrl.nodes`?

Comment (theefer) [inline kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js]: Ah maybe not, seems to be filtered down recursively? Not sure what it represents then :)

**Comment by theefer** [comment]
Great! :+1:

File: kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js (hunk 1 above)
Comment (tsop14) [inline kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js]: `nodeList` is a list of 'sibling' nodes, which is used when a collection is deleted. So, think it does need to be passed down recursively.

Comment (theefer) [inline kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js]: OK makes sense (might be nice to rename for clarity at some point, but not urgent)

**Pull request merged**